src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActions/SuggestedAction.cs

```diff
@@ -141,39 +141,55 @@ private async Task InvokeWorkerAsync(IProgressTracker progressTracker, Cancellat
         {
             await this.ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
 
-            IEnumerable<CodeActionOperation> operations = null;
+            object options = null;
             if (CodeAction is CodeActionWithOptions actionWithOptions)
             {
-                var options = actionWithOptions.GetOptions(cancellationToken);
-                if (options != null)
-                    operations = await GetOperationsAsync(actionWithOptions, options, cancellationToken).ConfigureAwait(true);
-            }
-            else
-            {
-                operations = await GetOperationsAsync(progressTracker, cancellationToken).ConfigureAwait(true);
+                using (Logger.LogBlock(FunctionId.CodeFixes_GetOptions, KeyValueLogMessage.Create(LogType.UserAction, m => CreateLogProperties(m)), cancellationToken))
+                {
+                    options = actionWithOptions.GetOptions(cancellationToken);
+                }
             }
 
-            AssertIsForeground();
-
-            if (operations != null)
+            using (Logger.LogBlock(
+                   FunctionId.CodeFixes_GetAndApplyChanges, KeyValueLogMessage.Create(LogType.UserAction, m => CreateLogProperties(m)), cancellationToken))
             {
-                // Clear the progress we showed while computing the action.
-                // We'll now show progress as we apply the action.
-                progressTracker.Clear();
-
+                IEnumerable<CodeActionOperation> operations = null;
                 using (Logger.LogBlock(
-                    FunctionId.CodeFixes_ApplyChanges, KeyValueLogMessage.Create(LogType.UserAction, m => CreateLogProperties(m)), cancellationToken))
+                       FunctionId.CodeFixes_GetChanges, KeyValueLogMessage.Create(LogType.UserAction, m => CreateLogProperties(m)), cancellationToken))
```

Comment:
>CodeFixes_GetChanges [](http://example.com/codeflow?start=34&length=20)

consider renaming this to `CodeFixes_GetAndApplyChanges_GetChanges` so its easier to disinguish it as a child of `CodeFixes_GetAndApplyChanges`